android/src/main/cpp/NativeProxy.cpp

```diff
@@ -94,6 +94,41 @@ jni::local_ref<NativeProxy::jhybriddata> NativeProxy::initHybrid(
       /**/);
 }
 
+#ifdef DEBUG
+void NativeProxy::checkVersion(jsi::Runtime &rnRuntime) {
+  std::string androidVersion;
+  try {
+    androidVersion =
+        getJniMethod<jstring()>("getReanimatedJavaVersion")(javaPart_.get())
+            ->toStdString();
+  } catch (std::exception &) {
+    throw std::runtime_error(
+        std::string(
+            "[Reanimated] Native side failed to resolve Java code version.\n") +
+        "See `http://localhost:3000/react-native-reanimated/docs/guides/troubleshooting#reanimated-native-side-failed-to-resolve-java-code-version` for more details.");
+  }
+  auto maybeJSVersion =
+      rnRuntime.global().getProperty(rnRuntime, "_REANIMATED_VERSION_JS");
+  if (maybeJSVersion.isUndefined()) {
+    throw std::runtime_error(
+        std::string(
+            "[Reanimated] Native side (Java) failed to resolve JS version of `react-native-reanimated`.\n") +
+        "See `http://localhost:3000/react-native-reanimated/docs/guides/troubleshooting#reanimated-native-side-java-failed-to-resolve-js-version-of-react-native-reanimated` for more details.");
+  }
+  auto JSVersion = maybeJSVersion.asString(rnRuntime).utf8(rnRuntime);
+  if (androidVersion != JSVersion) {
```

Comment:
Let's compare against C++ version, not JS version.
```suggestion
  if (javaVersion != cppVersion) {
```